src/components/elements/button/button.types.ts

```diff
@@ -0,0 +1,16 @@
+import type { Icon, IconWeight } from '@phosphor-icons/react';
+import type { AriaAttributes, AriaRole, HTMLAttributeAnchorTarget } from 'react';
+
+export interface ButtonProps extends AriaAttributes {
+  as?: 'button' | 'a' | 'Link';
+  role?: AriaRole;
+  target?: HTMLAttributeAnchorTarget;
+  variant?: 'primary' | 'secondary';
+  withBorder?: boolean;
+  small?: boolean;
+  type?: 'submit' | 'button' | 'reset';
+  onClick?: (event: MouseEvent) => void;
+  href?: string;
```

Comment:
Can we make the type stronger here? If the `as` value is `link` href should be required.